Source/JavaScriptCore/b3/B3CheckSpecial.cpp

```diff
@@ -101,9 +101,9 @@ CheckSpecial::~CheckSpecial()
 Inst CheckSpecial::hiddenBranch(const Inst& inst) const
 {
     Inst hiddenBranch(m_checkKind, inst.origin);
-    hiddenBranch.args.reserveInitialCapacity(m_numCheckArgs);
-    for (unsigned i = 0; i < m_numCheckArgs; ++i)
-        hiddenBranch.args.append(inst.args[i + 1]);
+    hiddenBranch.args.appendUsingFunctor(m_numCheckArgs, [&](size_t i) {
```

Comment:
Oh, I think I understand. In this case, for example, we could have calls the Vector constructor and done an assignment for example? If so, I agree in general. It would have the benefit of calling reserveInitialCapacity() instead of reserveCapacity() internally. In this particular case though, args is a `Vector<Arg, 3>`. The fact that it has an inline buffer means that the move assignment operator actually may actually copying (assuming the inline buffer is used) so there would be a cost from that if we didn't use appendUsingFunctor.

I need to look at other cases in this patch too.

Reply:
Following up in https://github.com/WebKit/WebKit/pull/20431.